### DISCUSSION TITLE
javax.xml.rpc-api-1.1.2.jar License in BIRT 4.9.0

### DISCUSSION THREAD

**Discussion by syc795** [eclipse-birt/birt#968] [Q&A] [answered]
The javax.xml.rpc jar uses the CDDL+GPL_1_1 license.
https://glassfish.dev.java.net/public/CDDL+GPL_1_1.html
common/javax.xml.rpc/javax.xml.rpc-api-1.1.2.jar/META-INF/maven/javax.xml.rpc/javax.xml.rpc-api/pom.xml 
Which license is used by the Birt 4.9.0?

**Comment by wimjongman** [comment, marked as answer]
https://www.eclipse.org/legal/epl-2.0/

**Comment by syc795** [comment]
Thanks for your answers, Wim.
I want to know how BIRT handles CDDL and GPL in javax.xml.rpc-api-1.1.2.jar.

**Reply by wimjongman** [reply]
You can probably find the answer here: https://www.eclipse.org/legal/epl-2.0/faq.php

**Reply by dm-ion** [reply]
Not a lawyer, but I presume that particular link would only be relevant if you were using jakarta.xml.rpc-api. You would have to double check but presumably 1.1.3 would be a drop in replacement for 1.1.2 (no namespace change). We have swapped out all javax used by BIRT for the jakarta variants (javax namespace still).

The nice thing about the eclipse license is: https://www.eclipse.org/legal/epl-2.0/faq.php#h.lgjcpvoq08a9 

```
[2.3. Is EPL-2.0 with the Secondary License clause considered dual licensing?](https://www.eclipse.org/legal/epl-2.0/faq.php#h.lgjcpvoq08a9)
It is extremely close to dual licensing. The EPL-2.0 is the only license, until such time as it is combined and distributed with a work under the Secondary License. After such time, any recipient of the combined work can consider the content licensed under the Secondary License. The original work remains under the EPL-2.0 and is never really dual-licensed. Once a downstream adopter has received the content under the Secondary License, they can modify and further distribute it solely under the terms of the Secondary License.

The EPL-2.0 file headers should remain on the source code content even if it has been made available under the Secondary License.
```

1.1.2 would still be under the oracle terms, not sure what their license requires.

**Reply by dm-ion** [reply]
Just to follow up 1.1.3 is indeed a drop in replacement. The diff is essentially just renaming maven artifacts, license changes, CI, etc.

1.1.2: https://github.com/javaee/javax.xml.rpc/commits/master
1.1.3: https://github.com/eclipse-ee4j/jax-rpc-api/releases/tag/1.1.3
